dev-docs/source/Standards/GUIStandards.rst

```diff
@@ -0,0 +1,666 @@
+a good idea, since it reduces the code per workflow algorithm. 
+However, long-term you now have N entry points to test separately and
+maintain.
+
+Having a single algorithm that handles the workflow associated with your GUI
+keeps it trivial to maintain, and also helps you maintain another standard;
+the workflow algorithm must work without the GUI.
+
+The GUI must work with workspace history
+----------------------------------------
+
+- Keep an eye out for reduction logic creeping into your MVP code
+- Any decisions the GUI makes about data (beyond browsing to it), goes into an alg.
+
+Any steps performed outside of the algorithm disappear from the workspace
+history. This breaks the reproducibility of data since running a script
+skirts around any code in your model.
+
+- Keep the workflow separate, the GUI should become a way to run a workflow
+- Run your workflow algorithm from the GUI, then re-run it as a script
+
+Imagine your GUI could be swapped for a CLI for users, would it still work on the back-end?
+
+Presenter
+=========
+
+.. _notifications_through_single_point:
+
+All notifications need to go through a single point
+---------------------------------------------------
+
+- Becomes obvious when it's starting to grow outwards
+- Prevents dead code paths, and multiple code paths to do the same thing
+- Makes debugging significantly easier
+
+Firstly, having a single method in your presenter handle every notification
+means you can put a single break-point when debugging. Imagine a future dev
+trying to figure out why that field in the GUI keeps changing, 
+but no breakpoints hit because you made another notification handler. Now
+imagine you're that future dev.....
+
+The other aspect is having a single method that's growing beyond 30 lines
+is painfully obvious. Maybe you have some dead code paths that need to be 
+trimmed out, maybe you are suffering from scope creep or maybe your notifier
+is doing model logic. In any case it's easier to spot now rather than 
+in 5 different places.
+
+Do not let Qt types in
+-----------------------
+
+- Keep non-standard types contained outside the view
+- Some debuggers can't see what these type values are
+- You then only have to think about one type instead of doing conversions everywhere
+
+To use stretch a metaphor, when you have a hammer suddenly everything looks like a
+nail: Many developers will use conversions (hammer) at the last second to 
+coerce their type (nail) into something "standard" which the rest of the 
+code-base uses at the point they need it. 
+
+You only notice something is wrong when you've put in so many conversions that
+it's either hard to keep track of what the type is at this point (Python), or
+you can't see into them debugging (C++). It dawns on you that the wooden chair
+you were building is 90% nails and you're not sure which ones hold it together and
+which are responsible for it being incredibly uncomfortable.
```

Comment:
I understand the metaphor until this sentence about the chair, it doesn't really follow on from what you said so I would just take it out.
```suggestion
you can't see into them debugging (C++).
```

Reply:
Gone, I was on the fence about it